src/subcommands/sign.rs

```diff
@@ -55,3 +86,18 @@ impl Sign {
         Ok(())
     }
 }
+
+fn hash_data(data: &[u8], alg: Hash) -> Result<Vec<u8>> {
+    let mut hasher: Box<dyn DynDigest> = match alg {
+        Hash::Sha224 => Box::from(sha2::Sha224::new()),
+        Hash::Sha256 => Box::from(sha2::Sha256::new()),
+        Hash::Sha384 => Box::from(sha2::Sha384::new()),
+        Hash::Sha512 => Box::from(sha2::Sha512::new()),
+        _ => {
+            err!("Hashing algorithm ({:?}) not supported", alg);
+            return Err(ToolErrorKind::NotSupported.into());
+        }
+    };
+    hasher.update(&data);
```

Comment:
Is there a maximal amount of data that one `update` can take at a time? As in, maybe for very large text to sign, we might want to do it 245 bytes per 256 bytes

Reply:
> Is there a maximal amount of data that one update can take at a time?

I don't think so, it's probably more to allow you to call it multiple times if you need to.